src/services/GLSL/vertex_shader_catalog.glsl

```diff
@@ -129,12 +138,75 @@ vec2 rotate2D(vec2 vector, float theta) {
     return mat2(cosTheta, -sinTheta, sinTheta, cosTheta) * vector * uScaleAdjustment;
 }
 
+vec4 cubic(vec4 A, vec4 B, vec4 C, vec4 D, float t) {
+    float t2 = t * t;
+    float t3 = t * t * t;
+    vec4 a = -A / 2.0 + (3.0 * B) / 2.0 - (3.0 * C) / 2.0 + D / 2.0;
+    vec4 b = A - (5.0 * B) / 2.0 + 2.0 * C - D / 2.0;
+    vec4 c = -A / 2.0 + C / 2.0;
+    vec4 d = B;
+
+    return a * t3 + b * t2 + c * t + d;
+}
+
+vec4 bicubicFilter(sampler2D texture2, vec2 P) {
```

Comment:
@panchyo0 any idea how we could prevent this sort of repetition between files? I guess glslify might help. Otherwise manually concatenating strings could work :thinking:

Reply:
glslify looks like the right tech for creating a utilities.glsl

Reply:
@panchyo0 I've made a separate issue for this: https://github.com/CARTAvis/carta-frontend/issues/1574